### ISSUE TITLE
Statement Descriptors for Stripe

### ISSUE THREAD

**Issue by ideadude** [strangerstudios/paid-memberships-pro#934]
Stripe allows you now to add a "statement descriptor" to a plan and/or charge. These must be at least 5 characters, but no more than 22, cannot use the special characters <, >, \, ', or ".

https://stripe.com/docs/connect/statement-descriptors

We can add it to the plan here: https://github.com/strangerstudios/paid-memberships-pro/blob/dev/classes/gateways/class.pmprogateway_stripe.php#L1590-L1598

That way it shows up for recurring orders as well.

Can we set this automatically? First 22 characters of the level name? Include the site name? Include the site name if the level name is less than 5 characters (e.g. Plus).

We can write a gist to use the pmpro_stripe_create_plan_array filter to add the descriptor and test it on our sites. We want to set it to "Paid Memberships Pro" and "Memberlite" based on the site.

**Comment by MichaelApproved** [comment]
Allowing for a custom statement descriptor would be helpful. Thankfully, this is partially possible using the filter for the plan creation. Using custom code, anyone can add a descriptor for charges **OTHER THAN** the first one, like this:

```
add_filter('pmpro_stripe_create_plan_array', 'stripe_plan_statement_descriptor');

public function stripe_plan_statement_descriptor($plan) {
	$plan['product']['statement_descriptor'] = "My product";

	return $plan;
}
```

_Note: Since [Stripe API 2018-02-05](https://stripe.com/docs/upgrades#2018-02-05) the statement_descriptor needs to be [put into the product](https://stripe.com/docs/api/products/object#product_object-statement_descriptor), instead of the plan._

However, a subscription's initial payment doesn't come through the plan, it comes through a one-time charge. So, we need to update the statement_descriptor for that initial one-time charge. 

The problem is the [create a payment intent code](https://github.com/strangerstudios/paid-memberships-pro/blob/dev/classes/gateways/class.pmprogateway_stripe.php#L2204) doesn't offer a filter that we can hook into for the parameters we need to edit. 

```
$payment_intent = Stripe_PaymentIntent::create( $params );
```

Luckily, this can be fixed with a simple edit, like this:

```
$payment_intent = Stripe_PaymentIntent::create( apply_filters( 'pmpro_stripe_create_payment_intent_array', $params ) );
```

With that filter, anyone can update the $param details using their own custom code, like this

```
add_filter('pmpro_stripe_create_payment_intent_array', 'stripe_payment_intent_custom_statement_descriptor');

public function stripe_payment_intent_custom_statement_descriptor($params) {
	$params['statement_descriptor'] = "My product";

	return $params;
}
```

**In summary, please add the filter to payment intent creation, so we can customize the stripe statement descriptor. Thanks!**

**Comment by ideadude** [comment]
We recently added a `pmpro_stripe_payment_intent_params` filter that can be used to add custom payment descriptors.

Here's a gist demonstrating how to do it: https://gist.github.com/ideadude/16983fdfa0da12fc40ef36d870f4cbd0
